### PR TITLE
Fix utf8 prefix index

### LINKED ISSUE #397: Prefix index implementation for utf8 string is incorrect
Related to https://github.com/pingcap/tidb/issues/7104

```sql
mysql> show create table t1;
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                         |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `name` varchar(12) DEFAULT NULL,
  KEY `pname` (`name`(12))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> insert into t1 values('借款策略集_网页');
Query OK, 1 row affected (0.14 sec)
```

```scala
scala> spark.sql("select * from t1 where name = '借款策略集_网页'").show
+----+
|name|
+----+
+----+

scala> spark.sql("select * from t1 where name < '借款策略集_网页'").show
+----+
|name|
+----+
|借款策略|
+----+
```

### PR DESCRIPTION
**Pull request by birdstorm**
fix #397 
This PR should be tested after https://github.com/pingcap/tidb/pull/7109 is merged.
Now blocked by https://github.com/pingcap/tidb/issues/7115
Will be merged after https://github.com/pingcap/tidb/pull/7194

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/IndexRangeBuilder.java [hunk 1]
```diff
@@ -137,7 +137,9 @@ private static void throwOnError(Expression node) {
           ranges.add(Range.singleton(literal));
           break;
         case NOT_EQUAL:
-          // Simply break here because an empty RangeSet will be evaluated as a full range
+          // Should return full range because prefix index predicate for NOT_EQUAL
+          // will be split into an NOT_EQUAL filter and a full range scan
+          ranges.add(Range.all());
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/IndexRangeBuilder.java]: This PR also finds a range build error here, see comment for details.

File: core/pom.xml [hunk 2]
```diff
@@ -269,7 +269,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>2.7</version>
+                <version>2.22.0</version>
```
Comment (ilovesoup) [inline core/pom.xml]: LGTM

**Pull request merged**